### ISSUE TITLE
Java: Using Custom Approval template with old spring security oauth library can lead to remote code execution

### ISSUE THREAD

**Issue by dogangcr** [github/securitylab#61]
## CVE ID(s)

- CVE-2018-1260

## Report

Spring Security OAuth, versions 2.3 prior to 2.3.3 and 2.2 prior to 2.2.2 and 2.1 prior to 2.1.2 and 2.0 prior to 2.0.15 and older unsupported versions, contains a remote code execution vulnerability. A malicious user or attacker can craft an authorization request to the authorization endpoint that can lead to a remote code execution when the resource owner is forwarded to the approval endpoint.

Spring security has OAuth solution for solve single sign on requirements. If developers use this library with old spring security oauth library and default approval page , developer will be affected from this vulnerability.

OAuth authorization code flow is like below. CVE-2018-1260 is focused on step 4. RCE attacks happened in step 4, if developer used default approval page. For prevent it , custom approval page or latest jar should be used.

![image](https://user-images.githubusercontent.com/12141715/77737475-1d887280-701f-11ea-8416-a7a1462fc769.png)

Related pull request:
Semmle/ql#3140

- [x] Are you planning to discuss this vulnerability submission publicly? (Blog Post, social networks, etc). *We would love to have you spread the word about the good work you are doing*
 Yes.

**Comment by xcorail** [comment]
Hello @dogangcr 

Thanks for the submission! We have reviewed your report and validated your findings. After internally assessing the findings and the query we have determined this query is not eligible for a reward under the Bug Bounty program for the reason explained in the PR. We only award higher quality queries as part of the program with attribution on our Bug Bounty site and a payout between $500 and $3,000.

Best regards and happy hacking!
